akka-actor/src/main/scala/akka/actor/ActorPath.scala

```diff
@@ -368,16 +368,26 @@ final class ChildActorPath private[akka] (val parent: ActorPath, val name: Strin
   }
 
   override def toStringWithAddress(addr: Address): String = {
-    val diff = addressStringLengthDiff(addr)
-    val length = toStringLength + diff
-    buildToString(new JStringBuilder(length), length, diff, _.toStringWithAddress(addr)).toString
+    if (IgnoreActorRef.isIgnoreRefPath(this)) {
```

Comment:
Is that really worth? 

It will mean that every single ActorPath will do this comparison at constructor time while now it's only done when we need it. 

Unless we make it a lazy val

Reply:
Most ActorPaths probably do not see any calls to either of these two methods, but I think paths that do are more likely to see several calls (called from serialization since  they are used in a remoting context). 

Lazy would be bad because that would be a synchronization on each access to the boolean, since it is the outcome of a calculation that never changes it could be lazy-calculated without synchronization/volatile but since when serializing there are likely a lot of other more expensive things going on so let's leave as is for now.